fix: Doctype query in create new shortcut

scmmishra opened the pull request:
While creating a desk shortcut, users were able to see the list of all doctypes. This would allow them to create the shortcut, however it won't be visible on the desk because of permission filetering while building desk.

![image](https://user-images.githubusercontent.com/18097732/98086223-76b56000-1ea4-11eb-8721-24a480f24de6.png)

This PR fixes this by using custom query for doctype from `permitted_documents_for_user.py`

<img width="627" alt="Screen Shot 2020-11-04 at 1 42 43 PM" src="https://user-images.githubusercontent.com/18097732/98086283-892f9980-1ea4-11eb-93cc-0084715eafac.png">

scmmishra committed:
fix: doctype query in create new shortcut

scmmishra commented:
@Mergifyio backport version-13-beta-pre-release

surajshetty3416 commented on frappe/public/js/frappe/widgets/widget_dialog.js:
```diff
@@ -150,7 +150,13 @@ class ShortcutDialog extends WidgetDialog {
+							return {
+								query: "frappe.core.report.permitted_documents_for_user.permitted_documents_for_user.query_doctypes",
+								filters: {
+									istable: false,
```
`istable: false` is not used in `query_doctypes`. You may need to update `query_doctypes` method if you need this filter.

scmmishra replied:
Ohh yeah, just saw that.. Updating

scmmishra replied:
@surajshetty3416 turns out that the `build_permissions` functions excludes child tables by default. I removed the filter param, and added a comment in the original function

scmmishra committed:
refactor: remove istable param

The query_doctypes function relies on UserPermissions build_permissions, which excludes child tables by default
